File: 899-binary-gap/binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        int last = -1;
        int maxGap = 0;

        for (int i = 0; n > 0; i++) {
            if (n & 1) {
                if (last != -1)
                    maxGap = max(maxGap, i - last);
                last = i;
            }
            n >>= 1;
        }

        return maxGap;
    }
};

File: 899-binary-gap/binary-gap.swift
class Solution {
    func binaryGap(_ n: Int) -> Int {
        var n = n
        var last = -1
        var maxGap = 0
        var position = 0
        
        while n > 0 {
            if (n & 1) == 1 {
                if last != -1 {
                    maxGap = max(maxGap, position - last)
                }
                last = position
            }
            n >>= 1
            position += 1
        }
        
        return maxGap
    }
}